MaterialLib/PorousMedium/UnsaturatedProperty/CapillaryPressure/BrookCoreyCapillaryPressure.h

```diff
@@ -0,0 +1,97 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * \file   BrookCoreyCapillaryPressure.h
+ *
+ *  Created on November 1, 2016, 9:45 AM
+ */
+
+#ifndef OGS_BROOK_COREY_CAPILLARY_PRESSURE_H
+#define OGS_BROOK_COREY_CAPILLARY_PRESSURE_H
+
+#include <array>
+#include <limits>  // std::numeric_limits
+
+#include "CapillaryPressureSaturation.h"
+
+namespace MaterialLib
+{
+namespace PorousMedium
+{
+/**
+ *   \brief Brook-Corey capillary pressure saturation model
+ *
+ *   \f[p_c=p_b S_e^{-1/m}\f]
+ *   with
+ *   \f[S_e=\dfrac{S-S_r}{S_{\mbox{max}}-S_r}\f]
+ *   where
+ *    \f{eqnarray*}{
+ *       &p_b&            \mbox{ entry pressure,}\\
+ *       &S_r&            \mbox{ residual saturation,}\\
+ *       &S_{\mbox{max}}& \mbox{ maximum saturation,}\\
+ *       &m(>=1) &        \mbox{ exponent.}\\
+ *    \f}
+ *
+ *    Note:
+ *     \f[m=1/(1-n)\f]
+ */
+class BrookCoreyCapillaryPressure final : public CapillaryPressureSaturation
+{
+public:
+    /** \param parameters An array contains the five parameters:
+     *                     [0] \f$ p_b \f$
+     *                     [1] \f$ S_r \f$
+     *                     [2] \f$ S_{\mbox{max}} \f$
+     *                     [3] \f$ m \f$
+     *                     [4] \f$ P_c^{\mbox{max}}\f$
+     */
+    BrookCoreyCapillaryPressure(std::array<double, 5> const& parameters)
+        : _pb(parameters[0]),
+          _Sr(parameters[1]),
+          _Smax(parameters[2]),
+          _mm(parameters[3]),
+          _Pc_max(parameters[4])
+    {
+    }
+
+    BrookCoreyCapillaryPressure(const BrookCoreyCapillaryPressure& orig)
```

Comment:
This can be shortened to
```c++
BrookCoreyCapillaryPressure(const BrookCoreyCapillaryPressure&) = default;
```

Also if copy constructor is present, add the move ctor, and maybe both assignments.